src/App.jsx

```diff
@@ -0,0 +1,133 @@
+import { useEffect, useState } from "react";
+import ProductTable from "./productTable";
+import SearchBar from "./searchBar";
+import SideBar from "./sideBar";
+import SortField from "./sort_field";
+
+function App() {
+  const [products, setProducts] = useState([])
+  const [categories, setCategories] = useState([])
+  const [types, setTypes] = useState([])
+  const [selectedTypes, setSelectedType] = useState([])
+  const [selectedCategory, setSelectedCategory] = useState([])
+  const [term, setTerm] = useState('')
+  const [prices, setPrices] = useState([])
+  const [selectedSort, setSelectedSort] = useState('asc')
+  const [selectedPage, setSelectedPage] = useState(1)
+  const [totalProduct, setTotalProduct] = useState('')
+  const [brands, setBrands] = useState([])
+  const [selectedBrand, setSelectedBrand] = useState([])
+  const debouncedSearchTerm = useDebounce(term, 500);
+
+  useEffect(() => {
+    const name = debouncedSearchTerm === '' ? '' : `name_like=${debouncedSearchTerm}&`;
+    const type = selectedTypes.length === 0 ? '' : typesPrameter(selectedTypes);
+    const category = selectedCategory.length === 0 ? '' : `category=${selectedCategory}&`;
+    const price = prices.length === 0 ? '' : `price_gte=${prices[0]}&price_lte=${prices[1]}&`;
+    const sort = `_sort=price&_order=${selectedSort}&`
+    const paginate = `_page=${selectedPage}&_limit=6&`
+    const brand = selectedBrand.length === 0 ? '' : `brand=${selectedBrand}&`
+    const url = 'http://localhost:3000/products?' + name + type + category + price + sort + brand + paginate;
```

Comment:
thay vì + thì e cho vào array xong join đi
```suggestion
    const url = 'http://localhost:3000/products?' + [name, type, categor,  price,  sort,  brand, paginate].join('');
```